tasks/initialise.yml

```diff
@@ -22,27 +22,78 @@
         PGSETUP_INITDB_OPTIONS: >-
           --encoding=UTF8 --locale=en_US.UTF-8 --auth-host=md5
 
-    - name: postgres | postgresql config file
-      template:
+    - name: postgres | Check for presence of "Modified by ome postgresql ansible role" in file
+      ansible.builtin.lineinfile:
+        path: "{{ postgresql_dist_confdir }}/postgresql.conf"
+        line: "#Modified by ome postgresql ansible role"
+        state: absent
+      check_mode: yes
+      changed_when: false
+      register: config_file_changed
+
+    - name: postgres | Check that the postgresql.conf.backup file exists
+      ansible.builtin.stat:
+        path: "{{ postgresql_dist_confdir }}/postgresql.conf.backup"
+      register: backup_result
+
+
+    # read the default postgresql configuration file
+    - name: postgres | get the postgres conf file contents
+      become_user: "{{ postgresql_become_user }}"
+      ansible.builtin.slurp:
+        src: "{{ postgresql_dist_confdir }}/postgresql.conf"
+      register: postgres_config_file_contents_o
+      when:
+        - not backup_result.stat.exists
+
+    # read the default postgresql configuration file
+    - name: postgres | get the postgres conf file contents from backup
+      become_user: "{{ postgresql_become_user }}"
+      ansible.builtin.slurp:
+        src: "{{ postgresql_dist_confdir }}/postgresql.conf.backup"
+      register: postgres_config_file_contents_b
+      when: backup_result.stat.exists
+
+    - set_fact: postgres_config_file_contents={{ postgres_config_file_contents_o }}
+      when:
+        - not backup_result.stat.exists
+
+    - set_fact: postgres_config_file_contents={{ postgres_config_file_contents_b }}
+      when:
+        - backup_result.stat.exists
+
+    - name: postgres | Copy a postgresql.conf to postgresql.conf.backup
+      become_user: "{{ postgresql_become_user }}"
+      ansible.builtin.copy:
+        src: "{{ postgresql_dist_confdir }}/postgresql.conf"
+        dest: "{{ postgresql_dist_confdir }}/postgresql.conf.backup"
```

Comment:
If I understand correctly the logic, this is meant to create a one-off copy of the configuration file assuming the latter has not originally been modified (and templated) by Ansible.

`.backup` implies to me that a copy of the modified configuration. I would if a different suffix e.g. `.orig` would better convey that this is meant to be the version of the configuration installed by the package manager